### ISSUE TITLE
set up user authentication (sign in, sign up, etc)

### ISSUE THREAD

**Issue by ornorth** [ChicoState/sound-space#41]
use firebase_auth to create an account registration/management system

**Comment by ornorth** [comment]
@snowmang1 If I could get some help/advice on this I would greatly appreciate it. I've set up a new branch 'auth_testing'. The code I have in 'account.dart' and 'authentication.dart' works insofar as it links to firebase and allows for setting up and logging in to an account. But it only works if I overwrite the widget in 'layout.dart' with the 'account.dart' code. I want the auth stuff to be contained to just the account page for now, but if I try then all that renders is a blank page.

TLDR I got firebase auth hooked up, but Flutter is giving me trouble

**Comment by snowmang1** [comment]
# Advice
- authentication file is to large move each stateful widget and description to its own file,
- I would make this a struct if possible so that you can explicitly specify type, this will make processing each element easier
``` dart
enum ApplicationLoginState {
 loggedOut,
 emailAddress,
 register,
 password,
 loggedIn,
}
```
- why is email conditional if it is required? [auth file: 28](https://github.com/ChicoState/sound-space/blob/fe62dc0e48fc2a5061dbd7155e9122c0ad26b17e/lib/pages/account/authentication.dart)
``` dart
final String? email;
```

# fixing process
- for future reference what I do is any widget that has to be built will get its own file
- all helper widgets can share a file
- Forms are much less complex when you use this file separation
- There is a problem with your switch statement at least on my end
- It may have something to do with the switch conditions type but I have not found the cause yet
sorry wasnt more help, I am not currently at home but Ill try to look over it some more tonight

**Comment by ornorth** [comment]
Splitting up the files seemed to fix everything. It now renders the login page exactly as I'd intended, so I think it's safe to say that the basics of firebase_auth are all set up. 

I've included comments addressing your other notes in my recent commit. I would argue that "ApplicationLoginState" doesn't need to have associated types because its only purpose is to track the login state of the current user, of which there can only be 1 at a time. It's essentially a 5-way boolean. As for "String? email", that field needs to be nullable in the case where the user is logged out. You could argue that it be initialized to an empty string instead, but that doesn't seem any cleaner than this solution.

If you have any other suggestions I'd love to hear them. For now, I think the next step is associating uploads with the user's account.
